sdk/management/docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,284 @@
+``` 
+
+In addition to this change, the **support for using auth file has been removed**. In old version, the user can choose to authenticate via the auth file, like this:
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+Azure azure = Azure.authenticate(new File("my.azureauth")).withDefaultSubscription();
+```
+**In new version, this feature has been removed.** If this creates concern on your side, please file an issue to let us know.
+
+For detailed information on the benefits of using the new authentication classes, please refer to [this page](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/management/docs/AUTH.md)
+
+## Customized Policy
+
+Because of adopting Azure Core which is a shared library across all Azure SDKs, there is also a minor change regarding how customized policy in configured. 
+
+In old version (`com.microsoft.azure-mgmt-**`), we use `withInterceptor` and pass the customized interceptor class to the Azure object 
+
+In new version (`com.azure.resourcemanager.**`), we use `WithPolicy` instead and pass the customized policy to the Azure object 
+
+So:
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+Azure azure = Azure.configure()
+    .withInterceptor(new CustomizedInterceptor())
+    .authenticate(credential)
+    .withDefaultSubscription();
+```
+
+**Equivalent in new version (`com.azure.resourcemanager.**`)**
+
+```java
+Azure azure = Azure.configure()
+    .withPolicy(new CustomizedPolicy())
+    .authenticate(credential, profile)
+    .withDefaultSubscription();
+```
+
+## Custom HTTP Client
+
+Similar to the customized policy, there are changes regarding how the custom HTTP client is configured as well. The re-designed HTTP client builder in the new version is more flexible and the user can choose their own implementation of HTTP client and plug in what they need into the configuration.
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+OkHttpClient.Builder builder = new OkHttpClient.Builder().proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 8888)));
+RestClient client = new RestClient.Builder(builder, new Retrofit.Builder())
+    .withCredentials(credential)
+    .build();
+
+Azure azure = Azure.authenticate(client, "<TenantId>")
+    .withDefaultSubscription();
+```
+
+**Equivalent in new version (`com.azure.resourcemanager.**`)**
+
+```java
+HttpClient client = new OkHttpAsyncHttpClientBuilder()
+    .proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress("127.0.0.1", 8888)))
+    .build();
+
+Azure azure = Azure.configure()
+    .withHttpClient(client)
+    .authenticate(credential, profile)
+    .withDefaultSubscription();
+```    
+    
+## Error Handling
+
+There is a minor namespace change in the exception class. To be specific, the previous `CloudException` has been re-named to `ManagementException`. 
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+final String resourceGroupName = random(8);
```

Comment:
My fault, I think it would be better to use `randomString()`, same as below. Or just remove these two lines.

In addition, you have removed the network creation in track 1, which would not raise exception actually.

Reply:
could you give an example where exception can occur using the same code for track 1 and track 2 ? otherwise this might cause confusion for user